x-pack/plugins/code/public/components/shortcuts/shortcuts_provider.tsx

```diff
@@ -123,11 +123,18 @@ class ShortcutsComponent extends React.Component<Props> {
       }
       return key === s.key;
     };
+
+    let isTriggered = false;
     for (const shortcut of this.props.shortcuts) {
       if (isPressed(shortcut) && shortcut.onPress) {
         shortcut.onPress(this.props.dispatch);
+        isTriggered = true;
       }
     }
+    if (isTriggered) {
+      // Discard this input since it's been triggered already.
+      event.preventDefault();
```

Comment:
@spacedragon we should prevent the default action if any shortcuts has been triggered.

Reply:
Normally, the `ShortcutsProvider` is the top level element. If it received an event, which means  
the child element didn't `preventDefault` it.

Reply:
just FYI, the case I am trying to fix here is when you press shortcut to focus the search bar input  box, the shortcut letter is placed into the input too.